src/main/java/org/opensearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,138 @@
+     * @param client                          ES node client that executes actions on the local node
+     * @param listener                        ES channel used to construct bytes / builder based outputs, and send responses
+     * @param anomalyDetectionIndices         anomaly detector index manager
+     * @param anomalyDetector                 anomaly detector instance
+     * @param requestTimeout                  request time out configuration
+     * @param maxSingleEntityAnomalyDetectors max single-entity anomaly detectors allowed
+     * @param maxMultiEntityAnomalyDetectors  max multi-entity detectors allowed
+     * @param maxAnomalyFeatures              max features allowed per detector
+     * @param method                          Rest Method type
+     * @param xContentRegistry                Registry which is used for XContentParser
+     * @param user                            User context
+     * @param searchFeatureDao                Search feature DAO
+     * @param typeStr                         specified type for validation
+     */
+    public ValidateAnomalyDetectorActionHandler(
+        ClusterService clusterService,
+        Client client,
+        ActionListener<ValidateAnomalyDetectorResponse> listener,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        AnomalyDetector anomalyDetector,
+        TimeValue requestTimeout,
+        Integer maxSingleEntityAnomalyDetectors,
+        Integer maxMultiEntityAnomalyDetectors,
+        Integer maxAnomalyFeatures,
+        RestRequest.Method method,
+        NamedXContentRegistry xContentRegistry,
+        User user,
+        SearchFeatureDao searchFeatureDao,
+        String typeStr
+    ) {
+        super(
+            clusterService,
+            client,
+            null,
+            listener,
+            anomalyDetectionIndices,
+            AnomalyDetector.NO_ID,
+            null,
+            null,
+            null,
+            anomalyDetector,
+            requestTimeout,
+            maxSingleEntityAnomalyDetectors,
+            maxMultiEntityAnomalyDetectors,
+            maxAnomalyFeatures,
+            method,
+            xContentRegistry,
+            user,
+            null,
+            searchFeatureDao,
+            true
+        );
+        String normalizedTypes = StringUtils.isBlank(typeStr) ? ValidationAspect.MODEL.getName() : typeStr.trim().replaceAll("\\s", "");
+        Set<String> typesInRequest = new HashSet<>(Arrays.asList(normalizedTypes.split(",")));
+
+        this.aspects = Sets
+            .union(DEFAULT_VALIDATION_ASPECTS, ValidationAspect.getNames(Sets.intersection(ALL_VALIDATION_ASPECTS_STRS, typesInRequest)));
+    }
+
+    @Override
+    public void start() throws IOException {
+        super.start();
+        if (aspects.contains(ValidationAspect.MODEL)) {
+            validateModelConfig();
+        }
+    }
+
+    // TODO: add implementation for model config validation
```

Comment:
Adding some more comments may be helpful here, like how all model config validation is non-blocker, maybe a link to this PR or other open issues about future plans to improve the validation